credentials/src/main/kotlin/web5/sdk/credentials/StatusListCredential.kt

```diff
@@ -0,0 +1,301 @@
+   *
+   * @param statusListCredentialId The id used for the resolvable path to the status list credential [String].
+   * @param issuer The issuer URI of the credential, as a [String].
+   * @param statusPurpose The status purpose of the status list cred, eg: revocation, as a [StatusPurpose].
+   * @param issuedCredentials The credentials to be included in the status list credential, eg: revoked credentials, list of type [VerifiableCredential].
+   * @return A [VerifiableCredential] instance.
+   *
+   * Example:
+   * ```
+   * val statusListCredential = StatusListCredential.create("http://example.com/statuslistcred/id123", "http://example.com/issuers/1", StatusPurpose.REVOCATION, listOf(vc1,vc2))
+   * ```
+   */
+  public fun create(
+    statusListCredentialId: String,
+    issuer: String,
+    statusPurpose: StatusPurpose,
+    issuedCredentials: List<VerifiableCredential>
+  ): VerifiableCredential {
+    val statusListIndexes = prepareCredentialsForStatusList(statusPurpose, issuedCredentials)
+    val bitString = bitstringGeneration(statusListIndexes)
+
+    val claims = mapOf(STATUS_PURPOSE to statusPurpose.toString().lowercase(), ENCODED_LIST to bitString)
+    val credSubject = CredentialSubject.builder()
+      .id(URI.create(statusListCredentialId))
+      .type("StatusList2021")
+      .claims(claims)
+      .build()
+
+    val vcDataModel = VcDataModel.builder()
+      .id(URI.create(statusListCredentialId))
+      .issuer(URI.create(issuer))
+      .issuanceDate(Date())
+      .context(URI.create("https://w3id.org/vc/status-list/2021/v1"))
+      .type("StatusList2021Credential")
+      .credentialSubject(credSubject)
+      .build()
+
+    return VerifiableCredential(vcDataModel)
+  }
+
+  /**
+   * Validates if a given credential is part of the status list represented by a [VerifiableCredential].
+   *
+   * @param credentialToValidate The [VerifiableCredential] to be validated against the status list.
+   * @param statusListCredential The [VerifiableCredential] representing the status list.
+   * @return A [Boolean] indicating whether the `credentialToValidate` is part of the status list.
+   *
+   * This function checks if the given `credentialToValidate`'s status list index is present in the expanded status list derived from the `statusListCredential`.
+   *
+   * Example:
+   * ```
+   * val isRevoked = validateCredentialInStatusList(credentialToCheck, statusListCred)
+   * ```
+   */
+  public fun validateCredentialInStatusList(
+    credentialToValidate: VerifiableCredential,
+    statusListCredential: VerifiableCredential
+  ): Boolean {
+    val statusListEntryValue: StatusList2021Entry =
+      StatusList2021Entry.fromJsonObject(credentialToValidate.vcDataModel.credentialStatus.jsonObject)
+
+    val statusLisCredStatusPurpose: String? =
+      statusListCredential.vcDataModel.credentialSubject.jsonObject[STATUS_PURPOSE] as? String?
+
+    requireNotNull(statusListEntryValue.statusPurpose)
```

Comment:
`requireNotNull` here doesn't seem right. Typical use would be for checking function preconditions (i.e. this will throw an `IllegalArgumentException`). Same for below.

Reply:
Is there anything we'd expect a caller to be able to do with the information from the checks here? Would it make more sense to just return false rather than make callers handle true/false & exception paths?

Reply:
It would alert the caller that they credentials they are passing in are not valid.

it will be true if the cred vs the credstatuslist is revoked
it will be false if the cred vs the credstatuslist is not revoked

it will throw an error if these are malformed creds.

I will fix up the exceptions to be more clear on that